goalgorilla/open_social
# Issue #3226672 by nkoporec: Overwrite title field for blocks does not work in Landing Page form

## Problem
When creating a new landing page and adding a new block in a section, the override title block field is not working correctly. The checkbox is enabled, but the actual field is not displayed, so currently you are not able to override the block title.

## Solution
The problem is with #states attribute. When rendering a form inside paragraphs, the '#name' attribute of fields are alternated since they are deeply nested within the paragraph fields. This means that the #states value of the original form(block settings form) is no longer correct, so we need to alter the form and set the correct #states attribute.

## Issue tracker
https://www.drupal.org/project/social/issues/3226672

## How to test

1. Enable social_landing_page module
2. Create a new landing page content type
3. Add a new section
4. Add a block reference
5. For 'Primary' field set 'Complete community activity stream'
6. Click the 'Override Title' checkbox and see that nothing happens (no additional field is displayed).

## Screenshots
Before:
![2021-08-04_12-12](https://user-images.githubusercontent.com/35064680/128164249-f8bbe47e-41be-45c5-9777-f9f25bd75334.png)

After:
![2021-08-04_12-13](https://user-images.githubusercontent.com/35064680/128164269-21217ffa-0f8e-4ca4-a1d8-0dc4c16459e5.png)


## Release notes
Fixed an issue where override title field would not work when rendered inside a paragraph of a landing page.
